File: BACKEND/BookMicroservice/src/main/java/com/example/bookmicroservice/entity/Book.java
package com.example.bookmicroservice.entity;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

/**
 * Clase que representa al objeto libro
 */
@Entity
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String nombre;
    private String descripcion;
    private int cantidad;
    private String editorial;
    private String imagen_url;
    private String dia_edicion;




    public Book() {
    }

    public Book(String nombre, String descripcion, int cantidad, String editorial, String imagen_url, String dia_edicion) {
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.cantidad = cantidad;
        this.editorial = editorial;
        this.imagen_url = imagen_url;
        this.dia_edicion = dia_edicion;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public int getCantidad() {
        return cantidad;
    }

    public void setCantidad(int cantidad) {
        this.cantidad = cantidad;
    }

    public String getEditorial() {
        return editorial;
    }

    public void setEditorial(String editorial) {
        this.editorial = editorial;
    }

    public String getImagen_url() {
        return imagen_url;
    }

    public void setImagen_url(String imagen_url) {
        this.imagen_url = imagen_url;
    }

    public String getDia_edicion() {
        return dia_edicion;
    }

    public void setDia_edicion(String dia_edicion) {
        this.dia_edicion = dia_edicion;
    }
}

File: FRONTEND/proyecto-bookstore/src/app/models/book.ts
/**
 * Esta clase representa al objeto de Libro de la libreria
 */
export class Book {
    /**Atributos de la clase */
    id?: number;
    nombre: string;
    descripcion: string;
    cantidad: number;
    editorial: string;
    imagen_url: string;
    dia_edicion: string;
    /**
     * Constructor de la clase
     * @param nombre 
     * @param descripcion 
     * @param cantidad 
     * @param editorial 
     * @param imagen_url 
     * @param dia_edicion 
     */
    constructor(nombre: string, descripcion: string, cantidad: number, editorial: string, imagen_url: string, dia_edicion: string) {
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.cantidad = cantidad;
        this.editorial = editorial;
        this.imagen_url = imagen_url;
        this.dia_edicion = dia_edicion;
    }
}

